DigitalExcellence/dex-backend
# Feature/312 call to actions

## Description

The call to actions implemented in the backend. Projects can contain call to actions and call to action options are added. These things are needed for the front-end for creating call to actions for a project.

## Type of change


## Checklist


## Steps to Test or Reproduce

1. Check the wiki and make sure that the updated scopes are noted.
2. Then run the project and make sure that both the unit and postman tests run (already checkable with Github actions, but it's never wrong the validate it again 😉 )
3. Validate that the CRUD functionalities for the call to action option endpoints work as expected (make sure you can't add twice the same option)
4. Add a project with a call to action and when retrieving that project, make sure that the call to action is available in the project and whenever you add a project, make sure you can only add an existing option to the project.

## Link to issue

Closes: #312
